### PR TITLE
Used to identify the Automatic Tank Gauge protocol 

### PR DESCRIPTION
**Pull request by wssxsxxsx**
Please sign (check) the below before submitting the Pull Request:

- [x] I have signed the ntop Contributor License Agreement at https://github.com/ntop/legal/blob/main/individual-contributor-licence-agreement.md
- [x] I have read the contributing guidelines at https://github.com/ntop/nDPI/blob/dev/CONTRIBUTING.md
- [x] I have updated the documentation (in doc/) to reflect the changes made (if applicable)

Link to the related [issue](https://github.com/ntop/nDPI/issues):

Describe changes:

### CODE REVIEW

File: src/lib/protocols/atg.c [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+						*ndpi_struct, struct ndpi_flow_struct *flow) {
+  ndpi_set_detected_protocol(ndpi_struct, flow, NDPI_PROTOCOL_ATG,
+			     NDPI_PROTOCOL_UNKNOWN, NDPI_CONFIDENCE_DPI);
+}
+
+
+static void ndpi_search_atg(struct ndpi_detection_module_struct *ndpi_struct,
+				    struct ndpi_flow_struct *flow) {
+  struct ndpi_packet_struct *packet = &ndpi_struct->packet;
+  int i=1;
```
Comment (IvanNardi) [inline src/lib/protocols/atg.c]: Remove this unused variable

**Review by IvanNardi — CHANGES_REQUESTED** [review]
Some changes:
* add `<ClCompile Include="..\src\lib\protocols\atg.c" />` to `windows/nDPI.vcxproj` (like you did in previous PR)
* update results files: `NDPI_FORCE_UPDATING_UTESTS_RESULTS=1 ./tests/do.sh` and "git add" all changed and new files

File: src/lib/protocols/atg.c (hunk 1 above)
Comment (wssxsxxsx) [reply src/lib/protocols/atg.c]: ok,I'll revise it right away

**Comment by IvanNardi** [comment]
LGTM. @utoni, @koltiradw, @0xA50C1A1 any comment?

**Comment by IvanNardi** [comment]
@wssxsxxsx, could you sign the CLA, please?

**Comment by wssxsxxsx** [comment]
@IvanNardi Is it to send the CLA by email？ I have already sent it.

**Comment by IvanNardi** [comment]
> @IvanNardi Is it to send the CLA by email？ I have already sent it.

No you don't, but I don't find your GitHub name (wssxsxxsx) in the list of the people who signed it. Could you double check, please?

**Comment by wssxsxxsx** [comment]
> > @IvanNardi Is it to send the CLA by email？ I have already sent it.
> 
> No you don't, but I don't find your GitHub name (wssxsxxsx) in the list of the people who signed it. Could you double check, please?

@IvanNardi Is it sent here-> legal@ntop.org

**Comment by 0xA50C1A1** [comment]
> LGTM. @utoni, @koltiradw, @0xA50C1A1 any comment?

I think it would be good to add some additional checks, in case this protocol can use a port different from 10001.

**Review by IvanNardi — APPROVED** [review]
I pushed some cosmetithic changes.
I think this PR is fine; we can always improve the detection later, if needed

**Comment by IvanNardi** [comment]
Thanks @wssxsxxsx for your work and your perseverance!
Thanks everyone for the reviews

**Pull request merged**